lua/packer.lua

```diff
@@ -167,6 +168,108 @@ packer.use_rocks = function(rock)
   end
 end
 
+-- Merges new into old; returns merged spec.
+local function merge_specs(old, new)
+  local name = new.short_name
+
+  -- disable: Warn & ignore on disabling a previously enabled plugin
```

Comment:
I may answer this question for myself further down the file, but: how is the old/new iteration order determined? If it's unstable, e.g. Lua table iteration order, then this could cause inconsistent behavior, since we only check the "old" version of the spec. That said, the inconsistency is only whether or not a warning gets printed, so it's not that big of a problem.

Reply:
Ah, this is probably a blind spot in my thinking, since I generally give packer several lists of plugin specs in the same order (and lists themselves are ordered), so I'm getting deterministic behavior.

I'll have to noodle on this a bit, since it may have an effect on the other options as well…

Reply:
I think we might actually be OK here - the only place this is called is within `manage`, and `manage` is only called within an `ipairs` loop, which should have a deterministic iteration order (since - afaik - plugins are always added with `use` in a deterministic order since they have to be given either as separate calls or in an arraylike table).